### PR TITLE
fix: theme switching, plus a little more

### PR DESCRIPTION
**Pull request by rpearce**
Resolves https://github.com/kutyel/kutyel-hakyll/issues/18

It is late here, so instead of writing a big description here, I will comment on the code, itself. That might be better, anyway :)

Try it, see how you like it, then read the code & comments!

— Robert

### CODE REVIEW

File: src/css/default.css [hunk 1]
```diff
@@ -1,3 +1,6 @@
+:root {
+  color-scheme: light, dark;
+}
```
Comment (rpearce) [inline src/css/default.css:3]: This lets the browser know that you support these color schemes

File: src/css/default.css [hunk 2]
```diff
@@ -62,7 +65,6 @@ body {
   background-color: var(--bgColor);
   color: var(--textColor);
   margin: 0;
-  transition: color 0.2s ease-out, background 0.2s ease-out;
```
Comment (rpearce) [inline src/css/default.css:65]: This transition is a core part of the problem

File: src/css/default.css [hunk 3]
```diff
@@ -34,7 +37,7 @@ body.light {
   --textLinkColor: #287cb4;
   --textLinkHoverColor: #206592;
 }
-body.dark {
+body[data-theme="dark"] {
```
Comment (rpearce) [inline src/css/default.css:40]: Working with a data attribute is easier than having to `document.body.classList.replace(..., ...)`

File: src/css/default.css [hunk 4]
```diff
@@ -133,43 +135,39 @@ figcaption {
...
+  color: var(--btnTextColor);
+  cursor: pointer;
   display: flex;
-  top: 1em;
-  right: 1em;
-  margin: 0;
-  padding: 0;
   font: inherit;
+  justify-content: center;
+  outline-offset: 0.1em;
```
Comment (rpearce) [inline src/css/default.css:152]: Actually, you should update this to `0.2em`

File: src/css/default.css [hunk 5]
```diff
@@ -133,43 +135,39 @@ figcaption {
...
+/* Theme switch */
 
-.toggle {
-  position: absolute;
+[data-theme-switch] {
+  -webkit-appearance: none;
+  align-items: center;
+  appearance: none;
+  background-color: var(--btnBgColor);
+  border: 1px solid var(--btnBgColor);
```
Comment (rpearce) [inline src/css/default.css:145]: Keep the `border` so that it shows up in Windows high contrast mode.

File: src/templates/default.html [hunk 6]
```diff
@@ -39,9 +39,57 @@
...
+          setTheme: (name) => {
+            document.body.setAttribute('data-theme', name)
+            localStorage.setItem('theme', name);
+            window.site.theme = name;
+            window.site.refreshThemeSwitch();
+
+            document.querySelector('iframe.giscus-frame')?.contentWindow.postMessage(
+              { giscus: { setConfig: { theme: name === 'dark' ? 'dark_dimmed' : 'light' } } },
+              'https://giscus.app'
+            );
```
Comment (rpearce) [inline src/templates/default.html:56]: I didn't verify if this works on the posts, so you might want to double check...it should! If it doesn't, I'm sure you can figure it out.

File: src/templates/default.html [hunk 7]
```diff
@@ -39,9 +39,57 @@
...
+      <span aria-hidden="true" data-theme-switch-icon="dark">🌙</span>
+      <span aria-hidden="true" data-theme-switch-icon="light">🌞</span>
+    </button>
+
     $body$
-    <script async src="./js/script.js"></script>
+
+    <script async>
+      ;(() => {
+        window.site.refreshThemeSwitch();
```
Comment (rpearce) [inline src/templates/default.html:86]: Once the page loads, this method will update the `<button>` to have the right `aria-label` and `aria-checked` values. If this isn't exactly how you want it, you could fade the button in via `opacity` (or `transform` from off the top of the page), and then any state change that it catches up on is handled while the element is transitioning in.

File: src/css/default.css (hunk 2 above)
Comment (kutyel) [reply src/css/default.css:65]: That's what I though, huge thanks! 🙌🏻

**Review by kutyel — APPROVED** [review]
Looks great, huge thanks for your help! 🙌🏻

**Pull request merged**